pkg/operations/rebuild_instance_inplace.go

```diff
@@ -341,6 +342,11 @@ func (inPlaceHelper *inplaceRebuildHelper) rebuildSourcePVCsAndRecreateInstance(
 		if err != nil {
 			return err
 		}
+		sourcePvc := &corev1.PersistentVolumeClaim{}
+		if err = cli.Get(reqCtx.Ctx, types.NamespacedName{Name: sourcePVCName, Namespace: tmpPVC.Namespace}, sourcePvc); err != nil {
+			// if not exists, wait for the pvc to recreate by external controller.
+			return err
```

Comment:
The implementation is not idempotent anyway: the source PVC will be forcibly deleted and then recreated in `recreateSourcePVC()`, but if errors occur during those steps, here it can't get the PVC anymore.

Reply:
if the pvc is deleted, it will be created by instanceSet controller.